### PR TITLE
Update release-notes to include tooling changes

### PR DESCRIPTION
**Pull request by AzeemMuzammil**
## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md [hunk 1]
```diff
@@ -87,6 +87,27 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
 
 ### Improvements
 
+#### CLI
+##### Refactor bal format command
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md:91]:
```suggestion
##### Refactored the `bal format` command
```

File: downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md [hunk 2]
```diff
@@ -87,6 +87,27 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
...
+Introduced new flags to `bal format` command and updated the format of the command. New `bal format` command would look like below,
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md:92]:
```suggestion
Introduced new flags to the `bal format` command and updated the format of the command. The new `bal format` command is shown below.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md [hunk 3]
```diff
@@ -87,6 +87,27 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
...
+
+`bal format [OPTIONS] [<package>|<module>|<source-file>]`
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md:94]:
```suggestion
`bal format [OPTIONS] [<package>|<module>|<source-file>]`
```
How the backticks are added is incorrect. Please check in other Learn pages. We have to have 3 backticks at the top and bottom to add a code snippet.

File: downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md [hunk 4]
```diff
@@ -87,6 +87,27 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
...
+
+     --file <file-name>
+        Format only a specific file in the Package
+    
+     -d, --dry-run
+        Perform a dry run of the formatter and see which files will
+        be formatted after the execution.
+```
+
+#### JSON to Record Converter
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md:108]:
```suggestion
#### JSON to record converter
```

File: downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md [hunk 5]
```diff
@@ -87,6 +87,27 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
...
+Improved the JSON to Record Converter tool to generate records with non-conflicting record names. The updated API requires the file URI and it will take care of the rest. When the API is used with Language-Server, the conversion will be much faster as the tool would use already existing compilation of the project.
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md:109]:
```suggestion
Improved the JSON to record converter tool to generate records with non-conflicting record names. The updated API requires only the file URI. When the API is used with the language server, the conversion will be faster as the tool would use the already-existing compilation of the project.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md [hunk 6]
```diff
@@ -87,6 +87,27 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
 
 ### Improvements
 
+#### CLI
+##### Refactored the `bal format` command
```
Comment (NipunaRanasinghe) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md:91]: @praneesha Shouldn't we ideally have the component names under H4 headers? (e.g `Formatter`, `JSON-to-record Converter`, etc)?

Comment (NipunaRanasinghe) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md:91]:
```suggestion
#### Refactored the `bal format` command
```

File: downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md [hunk 7]
```diff
@@ -87,6 +87,27 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
...
+Introduced new flags to the `bal format` command and updated the format of the command. The new `bal format` command is shown below.
```
Comment (NipunaRanasinghe) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md:92]: @AzeemMuzammil Shall reword this into something like below? WDYT?
```suggestion
- Improved the `bal format` CLI command by introducing new command options to selectively format modules or single files with a provided Ballerina package. The updated format of the `bal format` command is shown below.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md [hunk 8]
```diff
@@ -87,6 +87,27 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
...
+     --file <file-name>
+        Format only a specific file in the Package
+    
+     -d, --dry-run
+        Perform a dry run of the formatter and see which files will
+        be formatted after the execution.
+```
+
+#### Improved the JSON to record converter
+Improved the JSON to record converter tool to generate records with non-conflicting record names.
```
Comment (NipunaRanasinghe) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md:110]: @praneesha WDYT? 
```suggestion
Improved the JSON-to-record converter tool to be more context-aware, to be able to generate records with non-conflicting names.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md [hunk 9]
```diff
@@ -87,6 +87,27 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
...
+
+     --file <file-name>
+        Format only a specific file in the Package
+    
+     -d, --dry-run
+        Perform a dry run of the formatter and see which files will
+        be formatted after the execution.
+```
+
+#### Improved the JSON to record converter
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md:109]:
```suggestion
#### JSON to record converter
```

**Review by NipunaRanasinghe — APPROVED** [review]

**Comment by NipunaRanasinghe** [comment]
@AzeemMuzammil As the formatter improvements can be a possibly-breaking change, shall we mention it in the release notes itsel? 

@praneesha is there a dedicated place to mention such breaking changes in the release notes?

**Comment by praneesha** [comment]
> @AzeemMuzammil As the formatter improvements can be a possibly-breaking change for some use cases, shall we mention it in the release notes itsel?
> 
> @praneesha is there a dedicated place to mention such breaking changes in the release notes?

Yes, we will have a Breaking changes section as added in https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-2201.1.0#breaking-changes.

**Review by NipunaRanasinghe — APPROVED** [review]

**Comment by praneesha** [comment]
Breaking changes are added via https://github.com/ballerina-platform/ballerina-dev-website/pull/6313.

**Review by praneesha — APPROVED** [review]

**Pull request merged**